### ISSUE TITLE
Nuster Cache for S3 offload

### ISSUE THREAD

**Issue by serjponomarev** [jiangwenyuan/nuster#32]
Can i make Nuster cache for offload to S3 server?
I need fast write access for S3 client in one thread
For example S3 Client <--> Nuster Cache in RAM <--> [Minio](https://github.com/minio/minio)/[Ceph RGW](http://docs.ceph.com/docs/mimic/radosgw/)
if yes, how to do it? Config example, please

**Comment by jiangwenyuan** [comment]
Hi, just add your minio server as backend, and make minio client use the nuster ip:port.
But you might need to define some rules to only cache objects not the contents of buckets.

**Comment by serjponomarev** [comment]
Yes, right, i need cache only for objects, not for buckets.
What rules i need define in Nuster config for PUT/GET method?
Please, if its not hard for you, write config example for this:
cache size: 24Gb 
object size: 1Mb
ip:port minio: 127.0.0.1:9000
method: PUT/GET

**Comment by jiangwenyuan** [comment]
Hi, the hard part is how to tell what is a bucket list `mc ls /sth` and what is a object get `mc cp /sth sth`, as they use the same HTTP GET method `GET /sth`, unless you have some rules like all objects have a `.jpg` suffix, etc. Or you might need to observe the `mc --debug` command to find out if there is any special headers used in `ls` or `cp`, then you can use that to define nuster rules.

**Comment by serjponomarev** [comment]
if i have bucket in minio with name **bucket**, and object in this bucket with name **testobject**
What rules i need define in config?
I need only cache section or nosql section too?
Please, give me example

**Comment by serjponomarev** [comment]
Thanks for the example.
for the PUT method, is it enough to replace METH_GET with METH_PUT?
I need an example for the PUT method.

**Comment by serjponomarev** [comment]
For PUT requests, I used the solution here https://github.com/jiangwenyuan/nuster/issues/31
But there is a problem, after the first record, as I understand it, the existence of the record is checked and I see in the debug
**[CACHE] Checking key existence: NOT EXIST**
After the second lap of the PUT, I see
**[CACHE] Checking key existence: EXIST**
and i see acceleration.
How can I disable key verification?

**Comment by jiangwenyuan** [comment]
Hi, why do you want to cache PUT?

anyway

``` 
acl aa path /bucket/testobject 
nuster rule a if aa METH_GET or aa METH_PUT 
```

**Comment by jiangwenyuan** [comment]
what do you mean `disable key verification?`

key is used to locate the cache

**Comment by serjponomarev** [comment]
> Hi, why do you want to cache PUT?

I want to increase the speed of PUT requests.
Client -> PUT -> CACHE -> S3 backend

**Comment by serjponomarev** [comment]
@jiangwenyuan 
I can not understand why acceleration occurs after the second pass PUT requests, but not for first pass.
How decide this issue?

**Comment by serjponomarev** [comment]
> what do you mean disable key verification?
>key is used to locate the cache
I thought it was checking the existence of a record on the backend

**Comment by jiangwenyuan** [comment]
> I need a writeback cache for S3, is it possible to configure nuster like WB cache?
I don't think so, when you cache PUT request, only the response not the PUT body(testobject) is cached.
